File: src/main/java/technology/rocketjump/civimperium/mapgen/MapSettings.java
package technology.rocketjump.civimperium.mapgen;

public class MapSettings {

	public final StartEra startEra;
	public final MapType mapType;
	public final MapSize mapSize;
	public final WorldAge worldAge;
	public final SeaLevelSetting seaLevel;
	public final TemperatureSetting temperature;
	public final RainfallSetting rainfall;
	public final int numCityStates;
	public final int disasterIntensity;

	public MapSettings(StartEra startEra, MapType mapType, MapSize mapSize, WorldAge worldAge, SeaLevelSetting seaLevel,
					   TemperatureSetting temperature, RainfallSetting rainfall, int numCityStates, int disasterIntensity) {
		this.startEra = startEra;
		this.mapType = mapType;
		this.mapSize = mapSize;
		this.worldAge = worldAge;
		this.seaLevel = seaLevel;
		this.temperature = temperature;
		this.rainfall = rainfall;
		this.numCityStates = numCityStates;
		this.disasterIntensity = disasterIntensity;
	}
}

File: frontend/src/matches/MapSettings.js
import {List} from "semantic-ui-react";
import React from 'react';

const MapSettings = ({match}) => {

    return (
        <React.Fragment>
            <List horizontal>
                <List.Item>
                    <List.Content>
                        <List.Header>Num turns</List.Header>
                        {match.numTurns}
                    </List.Content>
                </List.Item>
                <List.Item>
                    <List.Content>
                        <List.Header>Start Era</List.Header>
                        {match.startEra}
                    </List.Content>
                </List.Item>
                <List.Item>
                    <List.Content>
                        <List.Header>Map Type</List.Header>
                        {match.mapType.replaceAll('_', ' ')}
                    </List.Content>
                </List.Item>
                <List.Item>
                    <List.Content>
                        <List.Header>Map Size</List.Header>
                        {match.mapSize}
                    </List.Content>
                </List.Item>
                <List.Item>
                    <List.Content>
                        <List.Header>World Age</List.Header>
                        {match.worldAge}
                    </List.Content>
                </List.Item>
                <List.Item>
                    <List.Content>
                        <List.Header>Sea Level</List.Header>
                        {match.seaLevel}
                    </List.Content>
                </List.Item>
                <List.Item>
                    <List.Content>
                        <List.Header>Temperature</List.Header>
                        {match.temperature}
                    </List.Content>
                </List.Item>
                <List.Item>
                    <List.Content>
                        <List.Header>Rainfall</List.Header>
                        {match.rainfall}
                    </List.Content>
                </List.Item>
                <List.Item>
                    <List.Content>
                        <List.Header>City States</List.Header>
                        {match.cityStates}
                    </List.Content>
                </List.Item>
                <List.Item>
                    <List.Content>
                        <List.Header>Disaster Intensity</List.Header>
                        {match.disasterIntensity}
                    </List.Content>
                </List.Item>
            </List>
            {match.startEraHint && <p><i>Hint: {match.startEraHint}</i></p>}
        </React.Fragment>
    );
}

export default MapSettings;